app/lib/actions/pulp/repository/copy_all_units.rb

```diff
@@ -2,16 +2,18 @@ module Actions
   module Pulp
     module Repository
       class CopyAllUnits < Pulp::AbstractAsyncTask
-        def plan(source_repo, _smart_proxy, target_repo, options = {})
+        def plan(target_repo, _smart_proxy, source_repos, options = {})
           filter_ids = options.fetch(:filters, nil)&.map(&:id)
           rpm_filenames = options.fetch(:rpm_filenames, nil)
           solve_dependencies = options.fetch(:solve_dependencies, false)
 
-          plan_self(source_repo_id: source_repo.id,
-                    target_repo_id: target_repo.id,
-                    filter_ids: filter_ids,
-                    solve_dependencies: solve_dependencies,
-                    rpm_filenames: rpm_filenames)
+          source_repos.each do |source_repo|
+            plan_self(source_repo_id: source_repo.id,
```

Comment:
We should check if we get a single repo here? I see one example here: https://github.com/Katello/katello/blob/c31deec0391d013f37b67364b9f5c0c859ad27bf/app/lib/actions/katello/repository/export.rb#L72

Reply:
oh right, plan_self won't work in that case!  i think i'll need to refactor out the run phase to another action.

Reply:
I updated this, but haven't tested my changes yet.  Will do so shortly